packages/thirdweb/src/wallets/interfaces/wallet.ts

```diff
@@ -74,6 +79,31 @@ export type Wallet<TWalletId extends WalletId = WalletId> = {
    * ```
    */
   getAccount(): Account | undefined;
+
+  /**
+   * @description
+   * Gets the linked profiles for the current wallet.
+   * This method is only available for in-app or ecosystem wallets.
+   *
+   * @returns An array of accounts user profiles linked to the current wallet.
+   *
+   * @example
+   * ```ts
+   * import { inAppWallet } from "@thirdweb-dev/wallets";
+   *
+   * const wallet = inAppWallet();
+   * wallet.connect({ strategy: "google" });
+   *
+   * const profiles = wallet.getProfiles();
+   *
+   * console.log(profiles[0].type);
+   * console.log(profiles[0].details.email);
+   * ```
+   */
+  getProfiles?: TWalletId extends "inApp" | EcosystemWalletId
+    ? () => Promise<Profile[]>
+    : undefined;
```

Comment:
thinking about this more, this feels _really_ weird to exist on this interface - can we instead do something similar that we do already with inapp user profile data and make it a standalone function you can call?

maybe we can pass the `wallet` into it and there it can be enforced that the wallet has to be inapp or ecosystem?

I really don't like the idea of adding this extra function on what is _supposed_ to be a fully generic interface for _every_ wallet. (Even if its hidden by wallet id typing (it'll still show up as an `undefined` key for every wallet)